crates/router/src/connector/iatapay/transformers.rs

```diff
@@ -291,8 +309,46 @@ fn get_iatpay_response(
     };
     let connector_response_reference_id = response.merchant_payment_id.or(response.iata_payment_id);
 
-    let payment_response_data = response.checkout_methods.map_or(
-        types::PaymentsResponseData::TransactionResponse {
+    let payment_response_data = match response.checkout_methods {
+        Some(checkout_methods) => {
+            let (connector_metadata, redirection_data) =
+                match checkout_methods.redirect.redirect_url.ends_with("qr") {
```

Comment:
Is this a reliable check?

Reply:
Yes. In all environments, response Url ends with their respective checkout method